src/server/mod.rs

```diff
@@ -349,46 +339,40 @@ where
         let codec = controlchan::FTPCodec::new();
         let (sink, stream) = codec.framed(tcp_tls_stream).split();
         let idle_session_timeout = self.idle_session_timeout;
-        let task = sink
-            .send(Reply::new(ReplyCode::ServiceReady, self.greeting))
-            .and_then(|sink| sink.flush())
-            .and_then(move |sink| {
-                sink.send_all(
-                    stream
-                        .map(Event::Command)
-                        .select(rx.map(Event::InternalMsg).map_err(|_| FTPErrorKind::InternalMsgError.into()))
-                        // Read events until we get an InternalMsg::Quit or the session times out
-                        .take_while(move |event| {
-                            if with_metrics {
-                                metrics::add_event_metric(&event);
-                            };
-                            match *event {
-                                Event::InternalMsg(InternalMsg::Quit) => Ok(false),
-                                _ => Ok(true),
-                            }
-                        })
-                        .and_then(event_handler_chain)
-                        .timeout(idle_session_timeout)
-                        .or_else(move |e| futures::future::ok(Self::handle_control_channel_error(e, with_metrics)))
-                        .map(move |reply| {
-                            if with_metrics {
-                                metrics::add_reply_metric(&reply);
-                            }
-                            reply
-                        })
-                        // Needed for type annotation, we can possible remove this once the compiler is
-                        // smarter about inference :)
-                        .map_err(|e: FTPError| e),
-                )
+        use futures03::compat::Sink01CompatExt;
+        use futures03::SinkExt;
+
+        let mut sink = sink.sink_compat();
+        sink.send(Reply::new(ReplyCode::ServiceReady, self.greeting)).await.unwrap();
+        sink.flush().await.unwrap();
```

Comment:
We should probably do something about these `unwrap` calls.

Reply:
Indeed

Reply:
Removed the unwraps in this file. I'll check others too.